src/helpers/enrich-dataset.ts

```diff
@@ -0,0 +1,186 @@
+
+export type HubSite = {
+    siteUrl: string,
+    portalUrl: string,
+    orgBaseUrl: string,
+    orgTitle: string
+};
+type FileType = 'shapefile' | 'csv' | 'geojson' | 'kml';
+
+export function enrichDataset(dataset: Record<string, any>, siteDetails: Record<string, string>): Record<string, any> {
+    // Download and Hub Links must be generated from Content
+    const datasetAttr = _.get(dataset, 'properties', {});
+    const content = datasetToContent({
+        id: dataset.id,
+        attributes: datasetAttr
+    } as DatasetResource);
+    const { siteUrl, portalUrl, orgBaseUrl, orgTitle } = siteDetails;
+    const { identifier, urls: { relative } } = content;
+    const additionalFields: Record<string, any> = {
+        ownerUri: getUserUrl({
+            portal: `${orgBaseUrl}/sharing/rest`,
+            username: datasetAttr.owner
+        } as UserSession) + '?f=json',
+        language: _.get(datasetAttr, 'metadata.metadata.dataIdInfo.dataLang.languageCode.@_value') || localeToLang(datasetAttr.culture) || '',
+        keyword: getDatasetKeyword(datasetAttr),
+        issuedDateTime: _.get(datasetAttr, 'metadata.metadata.dataIdInfo.idCitation.date.pubDate') || timestampToIsoDate(datasetAttr.created),
+        orgTitle,
+        provenance: _.get(datasetAttr, 'metadata.metadata.dataIdInfo.idCredit', ''),
+        hubLandingPage: concatUrlAndPath(siteUrl, relative.slice(1)),
+        downloadLink: concatUrlAndPath(siteUrl, `datasets/${identifier}`),
+        agoLandingPage: getAgoLandingPageUrl(dataset.id, portalUrl),
+        isLayer: isLayer(datasetAttr),
+        license: getDatasetLicense(datasetAttr)
+    };
+
+    const downloadLinkFor: (type: string) => string = getDownloadLinkFn(additionalFields.downloadLink, datasetAttr);
+
+    if (isProxiedCSV(datasetAttr)) {
+        additionalFields.accessUrlCSV = downloadLinkFor('csv');
+    }
+
+    if (isLayer(datasetAttr)) {
+        additionalFields.durableUrlGeoJSON = generateDurableDownloadUrl(dataset.id, siteUrl, 'geojson');
+        additionalFields.durableUrlCSV = generateDurableDownloadUrl(dataset.id, siteUrl, 'csv');
+        if (_.has(datasetAttr, 'layer.geometryType')) {
+            additionalFields.durableUrlKML = generateDurableDownloadUrl(dataset.id, siteUrl, 'kml');
+            additionalFields.durableUrlShapeFile = generateDurableDownloadUrl(dataset.id, siteUrl, 'shapefile');
+        }
+    }
+
+    if (datasetAttr.supportedExtensions?.includes(WFS_SERVER)) {
+        additionalFields.accessUrlWFS = ogcUrl(datasetAttr.url, 'WFS');
+    }
+
+    if (datasetAttr.supportedExtensions?.includes(WMS_SERVER)) {
+        additionalFields.accessUrlWMS = ogcUrl(datasetAttr.url, 'WMS');
+    }
+
+    dataset.properties = {
+        ...dataset.properties,
+        ...additionalFields
+    };
+
+    return dataset;
+};
+
+function generateDurableDownloadUrl(datasetId: string, siteUrl: string, fileType: FileType) {
+    const { itemId, layerId } = parseDatasetId(datasetId);
```

Comment:
This will break because `datasetId` is _not_ the hubId any more. It's the `item.id` which will never have a layer attached.

Reply:
`datasetId` is hubId for flatten response from OGC API. The provider will be used only using `flatten=true` query param.  ` (Example [here](https://santest-qa-pre-hub.hubqa.arcgis.com/api/search/v1/collections/all/items?flatten=true&id=68225aff18de4f13a47af7dcd4836103_1))